pkg/downloader/manager.go

```diff
@@ -218,13 +240,13 @@ func (m *Manager) Update() error {
 	return writeLock(m.ChartPath, lock, c.Metadata.APIVersion == chart.APIVersionV1)
 }
 
-func (m *Manager) loadChartDir() (*chart.Chart, error) {
-	if fi, err := os.Stat(m.ChartPath); err != nil {
-		return nil, fmt.Errorf("could not find %s: %w", m.ChartPath, err)
+func (m *Manager) loadChartDir(chartPath string) (*chart.Chart, error) {
+	if fi, err := os.Stat(chartPath); err != nil {
```

Comment:
The error where a remote repository is assumed to be a directory is coming from here. You could check the `file` protocol or absence of it to assume its a path

When updating your tests, you can setup a local remote repo to serve your charts like its done When adding a test to pull a dependency from a remote location, you can follow https://github.com/helm/helm/blob/d2236e95ed05e3c5b2598ceb677baa76011d9597/pkg/cmd/pull_test.go#L31

Reply:
https://github.com/helm/helm/blob/d2236e95ed05e3c5b2598ceb677baa76011d9597/pkg/downloader/manager.go#L633 is where a valid repo url is checked. You could extract that code to a helper function